### PR TITLE
Remove usage of deprecated std::bind2nd and std::bind1st function

### PR DESCRIPTION
**Pull request by Thomasb81**

### CODE REVIEW

File: cola/libtopology/topology_graph.h [hunk 1]
```diff
@@ -239,9 +239,7 @@ namespace topology {
          * this typedef can be used to declare a wrapper functor
          * for transferStraightConstraint
          */
-        typedef std::binder1st<
-            std::mem_fun1_t<void, Segment, StraightConstraint*> 
-            > TransferStraightConstraint;
+        typedef std::function<void(StraightConstraint*)> TransferStraightConstraint;
```
Comment (Thomasb81) [inline cola/libtopology/topology_graph.h:242]: libtopology/./../libtopology/topology_graph.h|242 col 22| warning: 'template<class _Operation> class std::binder1st' is deprecated: use 'std::bind' instead [-Wdeprecated-declarations]
||   242 |         typedef std::binder1st<
||       |                      ^~~~~~~~~
|| In file included from /usr/include/c++/13/bits/stl_function.h:1438,
/usr/include/c++/13/backward/binders.h|108 col 11| note: declared here
||   108 |     class binder1st
||       |           ^~~~~~~~~
libtopology/./../libtopology/topology_graph.h|243 col 18| warning: 'template<class _Ret, class _Tp, class _Arg> class std::mem_fun1_t' is deprecated [-Wdeprecated-declarations]
||   243 |             std::mem_fun1_t<void, Segment, StraightConstraint*>
||       |                  ^~~~~~~~~~
/usr/include/c++/13/bits/stl_function.h|1300 col 11| note: declared here
||  1300 |     class mem_fun1_t : public binary_function<_Tp*, _Arg, _Ret>
||       |           ^~~~~~~~~~
|| ../libtopology/topology_graph.h: In member function 'void topology::Edge::getTopologyConstraints(std::vector<topology::TopologyConstraint*>*) const':

File: cola/libtopology/topology_graph.h [hunk 2]
```diff
@@ -515,10 +513,10 @@ namespace topology {
         void getTopologyConstraints(std::vector<TopologyConstraint*>* ts) 
         const {
             forEach(
-                    std::bind2nd(
-                        std::mem_fun(&EdgePoint::getBendConstraint),ts),
-                    std::bind2nd(
-                        std::mem_fun(&Segment::getStraightConstraints),ts),
+                    std::bind(
```
Comment (Thomasb81) [inline cola/libtopology/topology_graph.h:516]: libtopology/./../libtopology/topology_graph.h|519 col 37| warning: 'std::mem_fun1_t<_Ret, _Tp, _Arg> std::mem_fun(_Ret (_Tp::*)(_Arg)) [with _Ret = void; _Tp = topology::EdgePoint; _Arg = vector<topology::TopologyConstraint*>*]' is deprecated: use 'std::mem_fn' instead [-Wdeprecated-declarations]
||   519 |                         std::mem_fun(&EdgePoint::getBendConstraint),ts),
||       |                         ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/13/bits/stl_function.h|1395 col 5| note: declared here
||  1395 |     mem_fun(_Ret (_Tp::*__f)(_Arg))
||       |     ^~~~~~~
libtopology/./../libtopology/topology_graph.h|518 col 33| warning: 'std::binder2nd<_Operation> std::bind2nd(const _Operation&, const _Tp&) [with _Operation = mem_fun1_t<void, topology::EdgePoint, vector<topology::TopologyConstraint*>*>; _Tp = vector<topology::TopologyConstraint*>*]' is deprecated: use 'std::bind' instead [-Wdeprecated-declarations]
||   518 |                     std::bind2nd(
||       |                     ~~~~~~~~~~~~^
||   519 |                         std::mem_fun(&EdgePoint::getBendConstraint),ts),
||       |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/13/backward/binders.h|172 col 5| note: declared here
||   172 |     bind2nd(const _Operation& __fn, const _Tp& __x)
||       |     ^~~~~~~
libtopology/./../libtopology/topology_graph.h|521 col 37| warning: 'std::const_mem_fun1_t<_Ret, _Tp, _Arg> std::mem_fun(_Ret (_Tp::*)(_Arg) const) [with _Ret = void; _Tp = topology::Segment; _Arg = vector<topology::TopologyConstraint*>*]' is deprecated: use 'std::mem_fn' instead [-Wdeprecated-declarations]
||   521 |                         std::mem_fun(&Segment::getStraightConstraints),ts),
||       |                         ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/13/bits/stl_function.h|1401 col 5| note: declared here
||  1401 |     mem_fun(_Ret (_Tp::*__f)(_Arg) const)
||       |     ^~~~~~~
libtopology/./../libtopology/topology_graph.h|520 col 33| warning: 'std::binder2nd<_Operation> std::bind2nd(const _Operation&, const _Tp&) [with _Operation = const_mem_fun1_t<void, topology::Segment, vector<topology::TopologyConstraint*>*>; _Tp = vector<topology::TopologyConstraint*>*]' is deprecated: use 'std::bind' instead [-Wdeprecated-declarations]
||   520 |                     std::bind2nd(
||       |                     ~~~~~~~~~~~~^
||   521 |                         std::mem_fun(&Segment::getStraightConstraints),ts),
||       |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/13/backward/binders.h|172 col 5| note: declared here
||   172 |     bind2nd(const _Operation& __fn, const _Tp& __x)
||       |     ^~~~~~~
|| topology_graph.cpp: In member function 'topology::Segment* topology::EdgePoint::prune(vpsc::Dim)':

**Comment by Thomasb81** [comment]
https://en.cppreference.com/w/cpp/utility/functional/bind12

std::bind1st and std::bind2nd are deprecated in C++11 ( and removed in C++17). Usage of those method create compilation warning.

**Comment by skieffer** [comment]
Looks good, thanks!

**Pull request merged**